Tested MYE run and completed Unit Testing

Arun-bioinformatics opened the pull request:
Included:
- testfiles: sample_sheet.csv, multiqc_data.json and config_example.yaml as part of unit testing.
- test_utils.py: script which unit tests all functions from utils.py
Updated:
- config fiels in idConfigFieldRelationship.json to make it capable of processing MYE runs.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/eggd_QC_Classifier/4)
<!-- Reviewable:end -->

Commit:
New repository: amplify config filed range and update json output
	modified:   full_assembly.py
    - modified json output to contain a key for thresholds for all samples
	modified:   resources/idConfigFieldRelationship.json
    - Added more elements to accommodate Config fields which are exclusive to MYE files.

Commit:
Commit with no change

Commit:
Starting with unit testing code.
	modified:   bin/utils.py
	modified:   full_assembly.py
	modified:   resources/idConfigFieldRelationship.json
	new file:   tests/__init__.py
	new file:   tests/test_utils.py
	renamed:    testfiles/CEN_multiqc_config_v2.1.0.yaml -> tests/testfiles/CEN_multiqc_config_v2.1.0.yaml
	new file:   tests/testfiles/config_example.yaml

Commit:
Expand on unit testing

Functions created for unit testing for reading config file, reading samplesheet and
get parameters.

	modified:   bin/utils.py
	modified:   tests/test_utils.py
	new file:   tests/testfiles/samplesheet_example.csv

Commit:
Unit Testing class TestGetMultiqcData included

	modified:   bin/utils.py
	modified:   tests/test_utils.py
	new file:   tests/testfiles/multiqc_data_example.json

Commit:
Included unit testing for get_status function

	modified:   bin/utils.py
    - modified the indentation for get_status, minor bug discovered
      during unit testing
	modified:   tests/test_utils.py
	modified:   tests/testfiles/config_example.yaml
    - minor modifications on config_example to accomodate tests.
	modified:   tests/testfiles/multiqc_data_example.json
    - added some keys to accomodate more tests.

Commit:
Include unit testing for get_output_filename function:
	modified:   tests/test_utils.py
	modified:   tests/testfiles/multiqc_data_example.json

pep8speaks commented:
Hello @Arun-bioinformatics! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`bin/utils.py`](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py):

> [Line 52:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L52): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 69:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L69): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 126:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L126): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 160:5](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L160): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '

* In the file [`full_assembly.py`](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/full_assembly.py):

> [Line 133:59](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/full_assembly.py#L133): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 134:51](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/full_assembly.py#L134): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'

* In the file [`tests/test_utils.py`](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py):

> [Line 15:1](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L15): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 21:29](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L21): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 23:52](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L23): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 24:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L24): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 24:73](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L24): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 25:36](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L25): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 36:44](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L36): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 37:33](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L37): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 53:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L53): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 53:86](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L53): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 54:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L54): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 54:87](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L54): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 55:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L55): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 56:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L56): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 57:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L57): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 58:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L58): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 59:29](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L59): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 59:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L59): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)
> [Line 61:1](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L61): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 97:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L97): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 105:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L105): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 113:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L113): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 120:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L120): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 127:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L127): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 145:57](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L145): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 155:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L155): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 163:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L163): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 172:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L172): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 180:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L180): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 187:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L187): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 189:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L189): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 197:55](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L197): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 198:64](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L198): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 198:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L198): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 199:62](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L199): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 199:73](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L199): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 199:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L199): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 200:53](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L200): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 202:68](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L202): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 202:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L202): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 203:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L203): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 204:43](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L204): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 205:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L205): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 205:95](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L205): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 217:67](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L217): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 217:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L217): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 220:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L220): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 220:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L220): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 79 characters)
> [Line 224:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L224): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 226:24](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L226): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 226:66](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L226): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 226:77](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L226): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 226:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L226): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 79 characters)
> [Line 229:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L229): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 229:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L229): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (99 > 79 characters)
> [Line 235:24](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L235): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 235:66](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L235): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 235:77](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L235): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 235:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L235): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 236:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L236): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 238:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L238): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 238:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L238): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 242:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L242): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 244:24](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L244): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 244:66](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L244): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 244:77](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L244): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 244:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L244): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 245:74](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L245): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 245:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L245): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 247:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L247): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 247:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L247): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 251:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L251): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 253:24](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L253): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 253:66](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L253): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 253:78](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L253): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 253:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L253): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 256:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L256): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 256:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L256): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 265:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L265): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 267:30](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L267): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 267:35](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L267): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 267:40](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L267): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 267:45](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L267): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 267:50](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L267): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 267:55](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L267): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 267:57](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L267): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 268:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L268): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 269:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L269): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 274:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L274): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 280:31](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L280): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 280:36](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L280): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 280:41](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L280): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 281:34](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L281): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 281:41](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L281): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 281:48](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L281): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 282:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L282): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 287:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L287): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 293:32](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L293): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 293:39](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L293): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 293:49](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L293): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 293:56](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L293): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 293:64](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L293): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 293:71](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L293): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 294:36](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L294): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 294:43](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L294): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 294:50](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L294): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 294:57](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L294): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 294:64](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L294): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 294:71](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L294): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 294:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L294): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 299:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L299): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 301:1](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L301): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 302:1](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L302): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 308:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L308): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 311:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L311): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 313:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L313): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 317:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L317): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 79 characters)
> [Line 319:80](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L319): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 319:81](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L319): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 322:25](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L322): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 324:1](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L324): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 324:12](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L324): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator

jethror1 reviewed (changes requested):
> Reviewed 8 of 12 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4)* status: 8 of 12 files reviewed, 15 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`bin/utils.py` line 51 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4dXg19_IwJyUtFuwg:-Ni4dXg19_IwJyUtFuwh:b-lj9c8f) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L51)):*
> ```Python
>     header_id = id_mapping.get(config_field)
> 
>     if header_id is None:
> ```

`if not header_id:` unless you need to explicitly check for None vs an empty string or 0 etc



_Code quote:_
```Python
 if header_id is None:
```
___
*[`bin/utils.py` line 69 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4djPh8hfm6Hu30YFz:-Ni4djPh8hfm6Hu30YG-:br3vefl) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L69)):*
> ```Python
> 
>     if not yaml_file.get("table_cond_formatting_rules"):
>         raise TypeError(f"{yaml_file} does not have 'table_cond_formatting_rules' key")
> ```

nice use of specific error type raising :thumbsup:



_Code quote:_
```Python
raise TypeError
```
___
*[`bin/utils.py` line 89 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4dpZAEECfJUQuih0v:-Ni4dpZAEECfJUQuih0w:b-774qlk) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L89)):*
> ```Python
>     parameters = yaml_file["table_cond_formatting_rules"][unique_id]
> 
>     return parameters
> ```

this would be nicer just as one line: `return yaml_file["table_cond_formatting_rules"][unique_id]`



_Code quote:_
```Python
    parameters = yaml_file["table_cond_formatting_rules"][unique_id]

    return parameters
```
___
*[`bin/utils.py` line 106 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4e05s3x4R-4mYxBMa:-Ni4e05s3x4R-4mYxBMb:b-58so6z) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L106)):*
> ```Python
>         sample_sheet = pandas.read_csv(file, header=None, usecols=[0])
>     column = sample_sheet[0].tolist()
>     print(column)
> ```

needed?



_Code quote:_
```Python
print(column)
```
___
*[`bin/utils.py` line 152 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4eE9L7KeCPEaNNsKL:-Ni4eE9L7KeCPEaNNsKM:b-15nk94) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L152)):*
> ```Python
>                 new_key = re.search(f"{sample_id}[A-Z0-9_]+", key)
> 
>             # Need to create an exception for "PCT_TARGET_BASES_20X"
> ```

is this still to do?



_Code quote:_
```Python
# Need to create an exception for "PCT_TARGET_BASES_20X"
```
___
*[`bin/utils.py` line 212 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4eIYY-2KsPe22_sHz:-Ni4eIYY-2KsPe22_sI-:bhyuxeo) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L212)):*
> ```Python
> 
>     if value == "unknown" or value == "warn":
>         status = "warn"
> ```

this is fine as is, but if you're wanting to check existence for more this can be written as:

```
if value in ["unknown", "warn", ...]:
```



_Code quote:_
```Python
    if value == "unknown" or value == "warn":
        status = "warn"
```
___
*[`tests/test_utils.py` line 75 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4ejlJ213-65E00bRm:-Ni4ejlJ213-65E00bRn:bwjg779) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L75)):*
> ```Python
>     def test_map_header_id(self):
>         """
>         Testing if outputs as expected
> ```

can this be more descriptive about what it is testing



_Code quote:_
```Python
Testing if outputs as expected
```
___
*[`tests/test_utils.py` line 226 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fA4w5yWMnpeynotL:-Ni4fA4w5yWMnpeynotM:bv6y6oo) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L226)):*
> ```Python
>         Test the function when header "PCT_TARGET_BASES_20X is processed as expected
>         """
>         tested_output =Classifier.get_key_value(TEST_MULTIQC_DATA,'sample_1','PCT_TARGET_BASES_20X')
> ```

spacing around `=` missing



_Code quote:_
```Python
tested_output =Cl
```
___
*[`tests/test_utils.py` line 235 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fG5f7Hd139BWZta6:-Ni4fG5f7Hd139BWZta7:b29ct5q) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L235)):*
> ```Python
>         Test the function when a given header ID is associated with Reads data.
>         """
>         tested_output =Classifier.get_key_value(TEST_MULTIQC_DATA,'sample_1','percent_duplicates')
> ```

spacing



_Code quote:_
```Python
=C
```
___
*[`tests/test_utils.py` line 244 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fJWp7PXp9XauDaPN:-Ni4fJWp7PXp9XauDaPO:b29ct5q) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L244)):*
> ```Python
>         Test the function when a given header ID is associated with the control data.
>         """
>         tested_output =Classifier.get_key_value(TEST_MULTIQC_DATA,'sample_1','Metric.Recall_snp')
> ```

spacing



_Code quote:_
```Python
t =C
```
___
*[`tests/test_utils.py` line 253 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fL9i2aaZ-jrV3_Wb:-Ni4fL9i2aaZ-jrV3_Wc:b29ct5q) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L253)):*
> ```Python
>         Test the function when data from the given header ID does not exist on given sample
>         """
>         tested_output =Classifier.get_key_value(TEST_MULTIQC_DATA,'sample_12','Metric.Recall_snp')
> ```

spacing



_Code quote:_
```Python
t =C
```
___
*[`tests/test_utils.py` line 274 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fOP5CBndWpWb0GfS:-Ni4fOP6BRG_kTAHwyVD:b-mfnx8k) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L274)):*
> ```Python
>             expected_output = expected_outcomes[index]
>             self.assertEqual(tested_output, expected_output,
>                              "The output of get_status with float numbers is not expected.")
> ```

I wouldn't have a test inside a loop, since if an early iteration fails the rest of the rest is not run. Consider changing this to build a list of errors, then assert that no errors are found like:

```
errors = []
for index, value in enumerate(tested_values):
    tested_output = Classifier.get_status(value, parameters)
    expected_output = expected_outcomes[index]
    if not tested_output == expected_output:
        errors.append(f"The output of get_status with float numbers is not expected with value {value}. Expected {expected_output}. Returned: {tested_output}")

if errors:
    raise AssertionError(errors)
```



_Code quote:_
```Python
        for index, value in enumerate(tested_values):
            tested_output = Classifier.get_status(value, parameters)
            expected_output = expected_outcomes[index]
            self.assertEqual(tested_output, expected_output,
                             "The output of get_status with float numbers is not expected.")
```
___
*[`tests/test_utils.py` line 287 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4gVIj2B-sSsRFIpqP:-Ni4gVIj2B-sSsRFIpqQ:bsvfxo9) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L287)):*
> ```Python
>             expected_output = expected_outcomes[index]
>             self.assertEqual(tested_output, expected_output,
>                              "The output of get_status with str values is not expected.")
> ```

as above with raising in a loop



_Code quote:_
```Python
        for index, value in enumerate(tested_values):
            tested_output = Classifier.get_status(value, parameters)
            expected_output = expected_outcomes[index]
            self.assertEqual(tested_output, expected_output,
                             "The output of get_status with str values is not expected.")
```
___
*[`tests/test_utils.py` line 299 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4gYL71kLWnEblyAW6:-Ni4gYL71kLWnEblyAW7:b-9chp70) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L299)):*
> ```Python
>             expected_output = expected_outcomes[index]
>             self.assertEqual(tested_output, expected_output,
>                              "The output of get_status with preset str values is not expected.")
> ```

as above



_Code quote:_
```Python
        for index, value in enumerate(tested_values):
            tested_output = Classifier.get_status(value, {})
            expected_output = expected_outcomes[index]
            self.assertEqual(tested_output, expected_output,
                             "The output of get_status with preset str values is not expected.")
```
___
*[`tests/test_utils.py` line 301 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4g_XQ8MLKoBiSNFf_:-Ni4g_XQ8MLKoBiSNFfa:b-58so6z) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L301)):*
> ```Python
>                              "The output of get_status with preset str values is not expected.")
> 
> #class TestGetOutputFilename():
> ```

needed?



_Code quote:_
```Python
#class TestGetOutputFilename():
```


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
Will generate html code coverage reports soon

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-:-NiFH7eT77ZLHRNXoqW3:b-wdrzgs)* status: 8 of 12 files reviewed, 15 unresolved discussions (waiting on @jethror1 and @mattgarner)

___
*[`bin/utils.py` line 51 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4dXg19_IwJyUtFuwg:-NiF2gxE6z4RIKVxu8R5:b9fh6v0) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L51)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

`if not header_id:` unless you need to explicitly check for None vs an empty string or 0 etc
</blockquote></details>

Generally speaking, the header_id variable is the output of a random_dictionary.get('non_existing_key'), which it's standard input is None. There could be rare cases where a key could be an empty string or a 0, so I decided to go with "header_id is None" to be very specific with my output.
___
*[`bin/utils.py` line 69 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4djPh8hfm6Hu30YFz:-NiF3YLzC3hSbGNyos5e:b-8fx2u7) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L69)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

nice use of specific error type raising :thumbsup:
</blockquote></details>

:+1: 
___
*[`bin/utils.py` line 89 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4dpZAEECfJUQuih0v:-NiF4-Nl8jCqrOMm8PfI:bb7lp6m) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L89)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

this would be nicer just as one line: `return yaml_file["table_cond_formatting_rules"][unique_id]`
</blockquote></details>

Ammended
___
*[`bin/utils.py` line 106 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4e05s3x4R-4mYxBMa:-NiF43p_7W5bDvg3kPku:bnu0gk) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L106)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

needed?
</blockquote></details>

No, don't think so
___
*[`bin/utils.py` line 152 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4eE9L7KeCPEaNNsKL:-NiF4DG-D8jkW-XTuMVC:bdsr8bl) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L152)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

is this still to do?
</blockquote></details>

It is done underneath, comment is changed to explain the intention of this line of code to reader.
___
*[`bin/utils.py` line 212 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4eIYY-2KsPe22_sHz:-NiFBxSVDq4sMdDmQNml:bzbo2zm) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/bin/utils.py#L212)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

this is fine as is, but if you're wanting to check existence for more this can be written as:

```
if value in ["unknown", "warn", ...]:
```
</blockquote></details>

Sorted
___
*[`tests/test_utils.py` line 75 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4ejlJ213-65E00bRm:-NiFC_wq70xGsXq7s5J6:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L75)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

can this be more descriptive about what it is testing
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 226 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fA4w5yWMnpeynotL:-NiFCYoBEMjTlVby7tim:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L226)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

spacing around `=` missing
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 235 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fG5f7Hd139BWZta6:-NiFH4mT2dCBClfVGRhL:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L235)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

spacing
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 244 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fJWp7PXp9XauDaPN:-NiFH5SH4EvHlJwnPUlr:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L244)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

spacing
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 253 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fL9i2aaZ-jrV3_Wb:-NiFH6-dEtQr4WKQgKMa:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L253)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

spacing
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 274 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4fOP5CBndWpWb0GfS:-NiFFlgu6H-IxjfwfxgB:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L274)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

I wouldn't have a test inside a loop, since if an early iteration fails the rest of the rest is not run. Consider changing this to build a list of errors, then assert that no errors are found like:

```
errors = []
for index, value in enumerate(tested_values):
    tested_output = Classifier.get_status(value, parameters)
    expected_output = expected_outcomes[index]
    if not tested_output == expected_output:
        errors.append(f"The output of get_status with float numbers is not expected with value {value}. Expected {expected_output}. Returned: {tested_output}")

if errors:
    raise AssertionError(errors)
```
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 287 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4gVIj2B-sSsRFIpqP:-NiFFmFwDRTlifrdfWzH:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L287)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

as above with raising in a loop
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 299 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4gYL71kLWnEblyAW6:-NiFFmiJ2W1IbVROkSfG:b-896fix) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L299)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

as above
</blockquote></details>

Done.
___
*[`tests/test_utils.py` line 301 at r2](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Ni4g_XQ8MLKoBiSNFf_:-NiFFrdnAADOHpdKcqPI:b32w12) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/9d823b393a75406edb77e742fefd88c89a2b8df4/tests/test_utils.py#L301)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

needed?
</blockquote></details>

No


<!-- Sent from Reviewable.io -->

Commit:
Test_utils.py correction, minor corections on working app
	modified:   bin/utils.py
	modified:   full_assembly.py
	modified:   tests/test_utils.py

jethror1 reviewed (approved):
> Reviewed 1 of 12 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @mattgarner)



<!-- Sent from Reviewable.io -->

Commit:
full_assembly.py unit testing script.
	modified:   full_assembly.py
	new file:   tests/test_full_assembly.py
	modified:   tests/test_utils.py
	modified:   tests/testfiles/multiqc_data_example.json
	modified:   tests/testfiles/samplesheet_example.csv

mattgarner reviewed (changes requested):
> Reviewed 7 of 12 files at r2, 1 of 3 files at r3, 2 of 5 files at r4.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4)* status: 10 of 13 files reviewed, 5 unresolved discussions (waiting on @Arun-bioinformatics and @jethror1)

___
*[`full_assembly.py` line 80 at r4](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-NjfgwTnBn2rRpzOFkvI:-NjfgwTnBn2rRpzOFkvJ:b-wwgcbi) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/full_assembly.py#L80)):*
> ```Python
> 
>     # List of config_fields
>     yaml_content = Classifier.read_config(args.config)
> ```

perhaps this should be `config_content` since the purpose is more important than the file type from which it came



_Code quote:_
```Python
yaml_content
```
___
*[`bin/utils.py` line 98 at r4](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-NjfjlEtCgcJ-9tRAyB-:-NjfjlEtCgcJ-9tRAyB0:b-c1fqol) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L98)):*
> ```Python
> 
>     Returns:
>         list: list of strings, each string represents a sample from multiqc.json 
> ```

represents in what way? Could this be clarified (that it returns sampleIDs)?



_Code quote:_
```Python
represents
```
___
*[`bin/utils.py` line 129 at r4](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-Njfk_rvA2lhDz0Esqdi:-Njfk_rvA2lhDz0Esqdj:bea9ank) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L129)):*
> ```Python
> 
> 
> def get_key_value(summarised_data, sample_id, header_id):
> ```

Could this be given a more specific name?



_Code quote:_
```Python
get_key_value
```
___
*[`bin/utils.py` line 129 at r4](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-NjfkjnR8nkjIiGfC1uf:-NjfkjnR8nkjIiGfC1ug:bvhy4dn) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L129)):*
> ```Python
> 
> 
> def get_key_value(summarised_data, sample_id, header_id):
> ```

Should this have a different name to describe its purpuse/meaning rather than its origin?



_Code quote:_
```Python
header_id
```
___
*[`bin/utils.py` line 145 at r4](https://reviewable.io/reviews/eastgenomics/QC_Classifier/4#-NjflWpjC3SD3QB8dhid:-NjflWpjC3SD3QB8dhie:b1102c1) ([raw file](https://github.com/eastgenomics/QC_Classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L145)):*
> ```Python
>     result = {}
>     for key, val in summarised_data.items():
>         if sample_id in key and header_id in key:
> ```

Perhaps add a comment to explain what key looks like? I assume it contains both the sample name and the metric name?
Could this ever go wrong by matching an unintended substring?



_Code quote:_
```Python
sample_id in key and header_id in key
```


<!-- Sent from Reviewable.io -->

Commit:
Minor changes in code for clarity

	modified:   bin/utils.py
	modified:   full_assembly.py

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_qc_classifier/4)* status: 10 of 13 files reviewed, 5 unresolved discussions (waiting on @jethror1 and @mattgarner)

___
*[`full_assembly.py` line 80 at r4](https://reviewable.io/reviews/eastgenomics/eggd_qc_classifier/4#-NjfgwTnBn2rRpzOFkvI:-NntQKpVDe2fW6cH0cUN:b-gdomhy) ([raw file](https://github.com/eastgenomics/eggd_qc_classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/full_assembly.py#L80)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

perhaps this should be `config_content` since the purpose is more important than the file type from which it came
</blockquote></details>

Sure, that will be changed
___
*[`bin/utils.py` line 98 at r4](https://reviewable.io/reviews/eastgenomics/eggd_qc_classifier/4#-NjfjlEtCgcJ-9tRAyB-:-NntQg7H9p_gsmqFVjpw:bml8fza) ([raw file](https://github.com/eastgenomics/eggd_qc_classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L98)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

represents in what way? Could this be clarified (that it returns sampleIDs)?
</blockquote></details>

will reworded to " each string is a sampleID from the multiqc.jso"
___
*[`bin/utils.py` line 129 at r4](https://reviewable.io/reviews/eastgenomics/eggd_qc_classifier/4#-Njfk_rvA2lhDz0Esqdi:-NntRXK63lFsY4ScDkxM:b-zi3l9a) ([raw file](https://github.com/eastgenomics/eggd_qc_classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L129)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Could this be given a more specific name?
</blockquote></details>

will change it to say get_sample_metric_value
___
*[`bin/utils.py` line 129 at r4](https://reviewable.io/reviews/eastgenomics/eggd_qc_classifier/4#-NjfkjnR8nkjIiGfC1uf:-NntS5DW2_ymr_ynxOOe:b-7pcslq) ([raw file](https://github.com/eastgenomics/eggd_qc_classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L129)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Should this have a different name to describe its purpuse/meaning rather than its origin?
</blockquote></details>

would the name metric_id be more self explanatory?
___
*[`bin/utils.py` line 145 at r4](https://reviewable.io/reviews/eastgenomics/eggd_qc_classifier/4#-NjflWpjC3SD3QB8dhid:-NnxQ0aq67VcNuMDC-FH:btonsiz) ([raw file](https://github.com/eastgenomics/eggd_qc_classifier/blob/25464175b0d9912914a73304cb0148f5afe9c607/bin/utils.py#L145)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Perhaps add a comment to explain what key looks like? I assume it contains both the sample name and the metric name?
Could this ever go wrong by matching an unintended substring?
</blockquote></details>

I have added a comment which presents what would the key structure would look like, here is the example: 
report_general_stat_data.0.sample_id.header_id


<!-- Sent from Reviewable.io -->

jethror1 commented:
@mattgarner have you more review you want to do on this, or can it be merged to continue with the next PR?

Commit:
PEP 8 fixes
	modified:   bin/utils.py
	modified:   full_assembly.py

Commit:
Minor PEP8 changes, unit testing files
	modified:   bin/utils.py
	modified:   tests/test_full_assembly.py
	modified:   tests/test_utils.py
